### ISSUE TITLE
Injection strategy Constructor does not generate compilable code

### ISSUE THREAD

**Issue by saw303** [mapstruct/mapstruct#1395]
Given the following mapper interface using Mapstruct 1.3.0-SNAPSHOT

```
@Mapper(config = SpringMappingConfig.class)
public interface MoneyMapper {

	default MonetaryAmount toMonetaryAmount(a.b.c.MonetaryAmount monetaryAmount) {
		return monetaryAmount != null ? monetaryAmount.toJavaMonetaryAmount() : null;
	}
}
```
and config 

```
@MapperConfig(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.ERROR, uses = {SimpleTypeMapper.class}, injectionStrategy = InjectionStrategy.CONSTRUCTOR)
public interface SpringMappingConfig {
}
```

the  result is:

```
@Component
public class MoneyMapperImpl implements MoneyMapper {

    @Autowired
    public MoneyMapperImpl(SimpleTypeMapper simpleTypeMapper) {
        this.simpleTypeMapper = simpleTypeMapper;
    }
}
```

The constructor parameter `simpleTypeMapper` is correctly assigned to `this.simpleTypeMapper` but there is no local `simpleTypeMapper`. 

*Update*
I have noticed that this feature works if all of the mappers are declared in the same module. In my case I compile a multi module project with Gradle and the `SimpleTypeMapper` is located in a different module. 

I will provide you an example project for that behaviour...

**Comment by filiphr** [comment]
@saw303 I think I know what the issue is. In MapStruct when the mapper in `uses` is not really used then no field is generated. We also need to omit the injection of that mapper in the constructor.


>I have noticed that this feature works if all of the mappers are declared in the same module.

This confuses me a bit. If the mapper is not actually used you should always get that error.

**Comment by filiphr** [comment]
I forgot to mention. Thanks a lot for trying out the latest master and reporting the issues.

btw. I think I have fixed the issue (see PR #1396). If you want you can try the PR by using jitpack 😄

**Comment by saw303** [comment]
Related to https://github.com/mapstruct/mapstruct/pull/1396
